### PR TITLE
ci: update github actions to latest stable

### PR DESCRIPTION
**Pull request by crazy-max**
Also adds dependabot to keep gha up to date

### CODE REVIEW

**Comment by tonistiigi** [comment]
Why draft?

**Comment by crazy-max** [comment]
> Why draft?

Provenance is now generated by default with bake action v4: https://github.com/tonistiigi/xx/actions/runs/7473571579/job/20337993559#step:6:15

Needs to handle this correctly first.

**Comment by crazy-max** [comment]
> > Why draft?
> 
> Provenance is now generated by default with bake action v4: https://github.com/tonistiigi/xx/actions/runs/7473571579/job/20337993559#step:6:15
> 
> Needs to handle this correctly first (align filename or just disable it).

I have disabled provenance for ld and ld64 jobs for now. Will look at this in follow-up as it's a bit tricky to handle renaming with the variant in the filename: https://github.com/crazy-max/xx/commit/2bfc62b4e77b376a01d696020d7453180f03c1d3

```
linux-386-ld-linux-386.tar.gz
linux-386-ld-linux-amd64.tar.gz
linux-386-ld-linux-arm64.tar.gz
linux-386-ld-linux-armv6.tar.gz
linux-386-ld-linux-armv7.tar.gz
linux-386-ld-linux-ppc64le.tar.gz
linux-386-ld-linux-s390x.tar.gz
provenance.linux_386.json
provenance.linux_amd64.json
provenance.linux_arm64.json
provenance.linux_arm_v6.json
provenance.linux_arm_v7.json
provenance.linux_ppc64le.json
provenance.linux_s390x.json
```

**Comment by crazy-max** [comment]
Should be good now: https://github.com/tonistiigi/xx/actions/runs/7501704227/job/20422870114#step:7:7

image

File: .github/workflows/ld.yml [hunk 1]
```diff
@@ -144,27 +150,37 @@ jobs:
...
       -
-        name: List artifacts
+        name: Rename provenance
         run: |
+          for pdir in ./dist/*/; do
+            (
+              cd "$pdir"
+              binname=$(find . -name '*.tar.gz')
+              filename=$(basename "${binname%.tar.gz}")
+              mv "provenance.json" "${filename}.provenance.json"
```
Comment (tonistiigi) [inline .github/workflows/ld.yml:177]: Can't we set the provenance filename with build request instead of renaming later?

Comment (crazy-max) [reply .github/workflows/ld.yml]: It works with https://github.com/tonistiigi/xx/pull/141/commits/5433bff9c48af90fbb87c4390d95ee692f2ca48d but filenames are not aligned so needs to renamed them anyway :persevere:: https://github.com/tonistiigi/xx/actions/runs/7783168263/job/21221059915?pr=141#step:7:7

image

I revert this change, I think my initial attempt looks better: https://github.com/tonistiigi/xx/pull/141#issuecomment-1889021879

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**